matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+            String voltageLevelId = containerMapping.busNumToVoltageLevelId.get(mBus.getNumber());
+            String substationId = containerMapping.voltageLevelIdToSubstationId.get(voltageLevelId);
+
+            // create substation
+            Substation substation = createSubstation(network, substationId);
+
+            // create voltage level
+            VoltageLevel voltageLevel = createVoltageLevel(mBus, voltageLevelId, substation, network, perUnitContext);
+
+            // create bus
+            createBus(mBus, voltageLevel);
+
+            // create load
+            createLoad(mBus, voltageLevel);
+
+            // create shunt compensator
+            createShuntCompensator(mBus, voltageLevel, perUnitContext);
+
+            //create generators
+            createGenerators(model, mBus, voltageLevel);
+        }
+    }
+
+    private static void createGenerators(MatpowerModel model, MBus mBus, VoltageLevel voltageLevel) {
+        model.getGenerators().stream().filter(gen -> gen.getNumber() == mBus.getNumber()).forEach(mGen -> {
+            String busId = getBusId(mGen.getNumber());
+            String genId = createGenId(mGen, voltageLevel.getNetwork());
+            Generator generator = voltageLevel.newGenerator()
+                    .setId(genId)
+                    .setConnectableBus(busId)
+                    .setBus(isInService(mGen) ? busId : null)
+                    .setTargetV(mGen.getVoltageMagnitudeSetpoint() * voltageLevel.getNominalV())
+                    .setTargetP(mGen.getRealPowerOutput())
+                    .setTargetQ(mGen.getReactivePowerOutput())
+                    .setVoltageRegulatorOn(true)
+                    .setMaxP(mGen.getMaximumRealPowerOutput())
+                    .setMinP(mGen.getMinimumRealPowerOutput())
+                    .add();
+
+            generator.newMinMaxReactiveLimits()
+                    .setMinQ(mGen.getMinimumReactivePowerOutput())
+                    .setMaxQ(mGen.getMaximumReactivePowerOutput())
+                    .add();
+
+            if ((mGen.getPc1() != 0) || (mGen.getPc2() != 0)) {
+                generator.newReactiveCapabilityCurve()
+                        .beginPoint()
+                        .setP(mGen.getPc1())
+                        .setMaxQ(mGen.getQc1Max())
+                        .setMinQ(mGen.getQc1Min())
+                        .endPoint()
+                        .beginPoint()
+                        .setP(mGen.getPc2())
+                        .setMaxQ(mGen.getQc2Max())
+                        .setMinQ(mGen.getQc2Min())
+                        .endPoint()
+                        .add();
+            }
+        });
+    }
+
+    private static Bus createBus(MBus mBus, VoltageLevel voltageLevel) {
+        String busId = getBusId(mBus.getNumber());
+        LOGGER.debug("Creating bus {}", busId);
+        Bus bus = voltageLevel.getBusBreakerView().newBus()
+                .setId(busId)
+                .setName(busId)
```

Comment:
To be consistent with generators (and maybe the rest of this importer), you should set the name for all the identifiables to the ID, or never set it.

Reply:
fixed: used a prefix, consistently